controller/gke-cluster-config-handler.go

```diff
@@ -0,0 +1,349 @@
+	}
+
+}
+
+func waitClusterRemoveExp(ctx context.Context, svc *gkeapi.Service, config *gkev1.GKEClusterConfig) (*gkeapi.Operation, error) {
+	var operation *gkeapi.Operation
+	var err error
+
+	for i := 1; i < 12; i++ {
+		time.Sleep(time.Duration(i*i) * time.Second)
+		operation, err = svc.Projects.Locations.Clusters.Delete(clusterRRN(config.Spec.ProjectID, config.Spec.Region, config.Spec.ClusterName)).Context(ctx).Do()
+		if err == nil {
+			return operation, nil
+		} else if !strings.Contains(err.Error(), "Please wait and try again once it is done") {
+			break
+		}
+	}
+	return operation, err
+}
+
+func getClientset(cluster *gkeapi.Cluster, ts oauth2.TokenSource) (kubernetes.Interface, error) {
+	capem, err := base64.StdEncoding.DecodeString(cluster.MasterAuth.ClusterCaCertificate)
+	if err != nil {
+		return nil, err
+	}
+	host := cluster.Endpoint
+	if !strings.HasPrefix(host, "https://") {
+		host = fmt.Sprintf("https://%s", host)
+	}
+
+	config := &rest.Config{
+		Host: host,
+		TLSClientConfig: rest.TLSClientConfig{
+			CAData: capem,
+		},
+		WrapTransport: func(rt http.RoundTripper) http.RoundTripper {
+			return &oauth2.Transport{
+				Source: ts,
+				Base:   rt,
+			}
+		},
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientset, nil
+}
+
+func generateClusterCreateRequest(config *gkev1.GKEClusterConfig) *gkeapi.CreateClusterRequest {
+	request := &gkeapi.CreateClusterRequest{
+		Cluster: &gkeapi.Cluster{
+			NodePools: []*gkeapi.NodePool{},
+		},
+	}
+	request.Cluster.Name = config.Spec.ClusterName
+	request.Cluster.InitialClusterVersion = *config.Spec.KubernetesVersion
+	request.Cluster.Description = config.Spec.Description
+	request.Cluster.EnableKubernetesAlpha = config.Spec.EnableAlphaFeature
+	request.Cluster.ClusterIpv4Cidr = config.Spec.ClusterIpv4Cidr
+	request.Cluster.EnableKubernetesAlpha = config.Spec.EnableAlphaFeature
+
+	disableHTTPLoadBalancing := config.Spec.ClusterAddons.HTTPLoadBalancing != nil && !*config.Spec.ClusterAddons.HTTPLoadBalancing
+	disableHorizontalPodAutoscaling := config.Spec.ClusterAddons.HorizontalPodAutoscaling != nil && !*config.Spec.ClusterAddons.HorizontalPodAutoscaling
+	disableNetworkPolicyConfig := config.Spec.ClusterAddons.NetworkPolicyConfig != nil && !*config.Spec.ClusterAddons.NetworkPolicyConfig
+	disableKubernetesDashboard := config.Spec.ClusterAddons.KubernetesDashboard != nil && !*config.Spec.ClusterAddons.KubernetesDashboard
+
+	request.Cluster.AddonsConfig = &gkeapi.AddonsConfig{
+		HttpLoadBalancing:        &gkeapi.HttpLoadBalancing{Disabled: !disableHTTPLoadBalancing},
+		HorizontalPodAutoscaling: &gkeapi.HorizontalPodAutoscaling{Disabled: disableHorizontalPodAutoscaling},
+		KubernetesDashboard:      &gkeapi.KubernetesDashboard{Disabled: disableKubernetesDashboard},
+		NetworkPolicyConfig:      &gkeapi.NetworkPolicyConfig{Disabled: disableNetworkPolicyConfig},
+	}
+
+	request.Cluster.InitialNodeCount = 1
```

Comment:
nit: Perhaps, this can move above?

Reply:
to be honest, this exists because the it has to have a value to work. and I had not updated the CRD yet as this is an initial skeleton of the project. I don't like it as is. will update to be read from the CRD. thanks for pointing this out. forgot it was in there. pretty lazy of me to leave it in... sorry was a lot of code.

Reply:
No problem :) Don't sweat it. Sounds like a plan.